### ISSUE TITLE
[Bug] SIM114 reports false positive finding 

### ISSUE THREAD

**Issue by kasium** [MartinThoma/flake8-simplify#70]
## Desired change

* **Rule(s)**: SIM114 
* **Adjustment**: Remove false positive findnings

## Explanation

The current one is wrong 😄 

## Example

```python
def complicated_calc(*arg, **kwargs):
    return 42

def foo(p):
    if p == 2:
        return complicated_calc(microsecond=0)
    elif p == 3:
        return complicated_calc(microsecond=0, second=0)
    return None
```

```
dummy.py:5:8: SIM114 Use logical or ((p == 2) or (p == 3)) and a single body
```

**Comment by oliver-sanders** [comment]
Here's another false positive:

```python
a = False      
b = True       
c = True       
               
if a:          
    z = 1      
elif b:        
    z = 2      
elif c:        
    z = 1 
```

The `a` and `c` branches cannot be merged into a single body because of the `b` branch inbetween.

**Comment by noamkush** [comment]
I noticed the buggy behavior happens on Python 3.6 but not on 3.8 (haven't checked 3.7).

**Comment by MartinThoma** [comment]
@oliver-sanders Uh, good catch! I haven't thought about something like this!

**Comment by ergoithz** [comment]
Also happening on 3.7.10

**Comment by MacFan4000** [comment]
https://github.com/MirahezeBots/sopel-channelmgnt/runs/4704103764?check_suite_focus=true we are running into the same issue.

**Comment by MartinThoma** [comment]
Fixed in `flake8-simplify==0.14.3`
